rackcity/models/asset.py

```diff
@@ -77,3 +78,28 @@ def save(self, *args, **kwargs):
                         self.asset_number = asset_number
                         break
             super(Asset, self).save(*args, **kwargs)
+            self.add_network_ports()
+            self.add_power_ports()
+
+    def add_network_ports(self):
+        from rackcity.models import NetworkPort
+        if len(NetworkPort.objects.filter(asset=self.id)) == 0:  # only add ports for new assets
+            model = self.model
+            for network_port_name in model.network_ports:
+                network_port = NetworkPort(
+                    asset=self,
+                    port_name=network_port_name,
+                )
```

Comment:
where are we going to handle the writing of MAC addresses to the NetworkPort table?

Reply:
Was planning to do the same place where I handle writing the connections, which would be in the function views after the asset has been saved

Reply:
swaggy